### DISCUSSION TITLE
Default values in typescript with enabled schema validation

### DISCUSSION THREAD

**Discussion by timtonk** [react-hook-form/react-hook-form#4524] [Q&A] [answered]
Hi there. I have a little issue with default values in TS.

I have a schema defined using `zod` where one field is marked as non-optional. Then I use the schema via `resolver` option in `useForm`, and also I'm trying to define default values there. Due to the fact, that the field is non-primitive (a complex object) I cannot define a reasonable default value for empty form (such as 0 for nums, an empty string for strings, etc), but at the same time I cannot set `null` as a default value, because TS deduced the type for the form from the schema described above.
Hence the question, how do I set `null` as a default value in this case?

Here is the sandbox with the extremely isolated error: https://codesandbox.io/s/heuristic-fast-twx14

**Comment by jorisre** [comment, marked as answer]
Hi @timtonk 

Zod is strongly typed and react-hook-form too. You can try to set the type yourself:

```ts
resolver: zodResolver<any>(schema) // Replace any by your type
```

Hope it helps

**Reply by timtonk** [reply]
Seems like it does the job, even though I feel like types for default values should accept nulls. But anyway thank you!
